google/cloud/storage/bucket.py

```diff
@@ -2957,6 +2957,31 @@ def object_retention_mode(self):
         if object_retention is not None:
             return object_retention.get("mode")
 
+    @property
+    def hierarchical_namespace_enabled(self):
+        """Whether hierarchical namespace is enabled for this bucket.
+
+        :setter: Update whether hierarchical namespace is enabled for this bucket.
+        :getter: Query whether hierarchical namespace is enabled for this bucket.
+
+        :rtype: bool
+        :returns: True if enabled, else False.
+        """
+        hns = self._properties.get("hierarchicalNamespace", {})
+        return hns.get("enabled")
+
+    @hierarchical_namespace_enabled.setter
+    def hierarchical_namespace_enabled(self, value):
+        """Enable or disable hierarchical namespace at the bucket-level.
+
+        :type value: convertible to boolean
+        :param value: If true, enable hierarchical namespace for this bucket.
```

Comment:
Currently can only be enabled at bucket creation and can't be changed at least not right now after bucket is created.

Reply:
Right, the tests reflect that. I've added a note in the docstrings.